packages/flutter_tools/test/src/context.dart

```diff
@@ -135,6 +135,7 @@ void testUsingContext(
           Pub: () => ThrowingPub(), // prevent accidentally using pub.
           CrashReporter: () => MockCrashReporter(),
           TemplateRenderer: () => const MustacheTemplateRenderer(),
+          BotDetector: () => const AlwaysFalseBotDetector(),
```

Comment:
This is probably okay, but I'm wondering why it's being added in this PR?

Reply:
It was to change the default behavior.

I instead made the `checks that Flutter installation is up-to-date` test have an explicit `false` bot fake.